docs/standards/generic-standards/02-lsp1-universal-receiver.md

```diff
@@ -15,34 +15,47 @@ sidebar_position: 3.2
+- bytes `data`: Any arbitrary data.
 
-The **universalReceiver** function emits an event with the data passed to it along with some additional data. It could be customized to **revert on calls**, this way the smart contract implementing it won't be able to receive **assets, information**, etc ..
+The `universalReceiver(...)` function emits an event with the data passed to it and some additional data. The function can then implement custom logic to make the contract behave differently based on the data received. For instance, the universalReceiver(...) function could:
 
-> Receiving contracts should take the `typeId` into consideration to properly **decode the data**.
+- revert on calls to completely disallow the smart contract from receiving assets, information, etc. :x:
+- register the received assets inside the contract storage (see [LSP5 - Received Assets](../universal-profile/06-lsp5-received-assets.md)) :heavy_plus_sign:
 
-![schema of universal receiver transaction](../../../static/img/ur-transaction.jpeg)
+> Receiving contracts should consider the `typeId` parameter to **decode the data correctly**.
 
-Smart contracts that are willing to implement this function are well advised **to register** the **[LSP1UniversalReceiver InterfaceId](../smart-contracts/interface-ids.md)** **using ERC165** to let other contracts know that they support this standard.
+![schema of universal receiver transaction](../../../static/img/ur-transaction.jpeg)
 
 ## Extension
 
-An **optional extension** to this standard is proposed to let contracts not just be notified on the incoming and outcoming transactions but to delegate their response to an external upgradeable contract that could **handle and react to certain calls** with custom logic.
+:::info
 
-This is possible through setting the address of the **external contract** as value for the **LSP1UniversalReceiver key** below, in the **[ERC725Y key value store](https://github.com/ERC725Alliance/erc725/blob/main/docs/ERC-725.md#erc725y)** of the contract implementing the **universalReceiver** function.
+See the **[LSP1-UniversalReceiverDelegate](../universal-profile/02-lsp1-universal-receiver-delegate.md)** standard for more details.
+
+:::
+
+LSP1UniversalReceiverDelegate is an **optional extension** to the LSP1 Standard. As well as notifying a contract about the incoming and outgoing transactions via an event, it will delegate the `universalReceiver(...)` functionality to an external contract that can **handle and react to specific calls** with its custom logic.
+
+To enable this optional extension, the address of the **external contract** MUST be set as a value for the **LSP1UniversalReceiver key** shown below. This key-value pair inside the **[ERC725Y key value store](https://github.com/ERC725Alliance/erc725/blob/main/docs/ERC-725.md#erc725y)** of the contract implementing the `universalReceiver(...)` function will act as a reference, making this external contract upgradeable if required.
```

Comment:
I made a mistake here when writing this section
It should be LSP1UniversalReceiverDelegate Key instead of LSP1UniversalReceiver Key